examples/maze-solver/maze.py

```diff
@@ -0,0 +1,204 @@
+                directions = {get_label(i, j, 'n'), get_label(i, j, 'w'), get_label(i+1, j, 'n'),
+                              get_label(i, j+1, 'w')}
+                self.csp.add_constraint(sum_to_two_or_zero, directions)
+
+    def _set_start_and_end(self):
+        """Sets the values of the start and end locations of the maze.
+        """
+        self.csp.fix_variable(self.start, 1)  # start location
+        self.csp.fix_variable(self.end, 1)  # end location
+
+    def _set_borders(self):
+        """Sets the values of the outer border of the maze; prevents a path from forming over the border.
+        """
+        for j in range(self.n_cols):
+            top_border = get_label(0, j, 'n')
+            bottom_border = get_label(self.n_rows, j, 'n')
+
+            try:
+                self.csp.fix_variable(top_border, 0)
+            except ValueError:
+                if not top_border in [self.start, self.end]:
+                    raise ValueError
+
+            try:
+                self.csp.fix_variable(bottom_border, 0)
+            except ValueError:
+                if not bottom_border in [self.start, self.end]:
+                    raise ValueError
+
+        for i in range(self.n_rows):
+            left_border = get_label(i, 0, 'w')
+            right_border = get_label(i, self.n_cols, 'w')
+
+            try:
+                self.csp.fix_variable(left_border, 0)
+            except ValueError:
+                if not left_border in [self.start, self.end]:
+                    raise ValueError
+
+            try:
+                self.csp.fix_variable(right_border, 0)
+            except ValueError:
+                if not right_border in [self.start, self.end]:
+                    raise ValueError
+
+    def _set_inner_walls(self):
+        """Sets the values of the inner walls of the maze; prevents a path from forming over an inner wall.
+        """
+        for wall in self.walls:
+            self.csp.fix_variable(wall, 0)
+
+    def get_bqm(self, penalty_per_tile=0.5):
+        """Applies the constraints necessary to form a maze and returns a BQM that would correspond to a valid path
+        through said maze.
+
+        Note: If penalty_per_tile is too large, the path will be too heavily penalized and the optimal solution might
+          no path at all.
+
+        Args:
+            penalty_per_tile: A number. Penalty for each tile that is included in the path; encourages shorter paths.
+
+        Returns:
+            A dimod.BinaryQuadraticModel
+        """
+        # Apply constraints onto self.csp
+        self._apply_valid_move_constraint()
+        self._set_start_and_end()
+        self._set_borders()
+        self._set_inner_walls()
+
+        # Grab bqm constrained for valid solutions
+        bqm = dbc.stitch(self.csp)
+
+        # Edit bqm to favour optimal solutions
+        for v in bqm.variables:
+            # Ignore auxiliary variables
+            if isinstance(v, str) and match("aux\d+$", v):
```

Comment:
Also, isn't there a more robust (future-proof) way of checking which variable is auxiliary? Stitch surely knows.